kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -139,11 +157,13 @@ public void onFailure(IOException e, Response response) {
           Status status = OperationSupport.createStatus(response);
           logger.error("Exec Failure: HTTP:" + status.getCode() + ". Message:" + status.getMessage(), e);
           if (!started.get()) {
+            queue.clear();
```

Comment:
This is an example of code (written by me) that assumes that onFailure() can only occur during open.

Still I am not sure if it makes sense to clear the queue. After all the queue is used like a CountDownLatch(1) that you can pass an exception to the guy that waits.